extensions/spring-xd-extension-http/src/main/java/org/springframework/integration/x/http/NettyHttpInboundChannelAdapter.java

```diff
@@ -210,6 +257,8 @@ protected void doStop() {
 		if (bootstrap != null) {
 			bootstrap.shutdown();
 		}
+		Map<String, Object> pathToProducer = getModuleState();
+		pathToProducer.remove(path);
```

Comment:
This needs to be symmetrical with the `put()`, which is currently done in the constructor.

We could implement `DisposableBean` and do this in the `destroy()` method, but it would probably be better to move the `ensurePathIsUnique()` logic (`put`) to `doStart()`.